lib/multi_level_request_logger/plug.ex

```diff
@@ -0,0 +1,57 @@
+defmodule ElixirTools.MultiLevelRequestLogger.Plug do
+  use Plug.Builder, init_mode: :runtime
+
+  require Logger
+
+  plug(Plug.Logger, builder_opts())
+
+  @default_log_level :info
+  @allowed_log_levels [:warn, :error, :info, :debug]
+
+  @impl true
+  @spec init(Plug.opts()) :: Plug.opts()
+  def init(opts), do: opts
+
+  @impl true
+  def call(conn, _opts \\ []) do
+    log_level = conn |> Map.get(:request_path) |> log_level_for_path()
+
+    super(conn, log: log_level)
+  end
+
+  defp log_level_for_path(path) do
```

Comment:
`spec` missing

Reply:
hm, super-strange that it was not detected by https://github.com/digitalorigin/pg-elixir-tools/blob/master/.circleci/config.yml#L76